reinforcement_learning/bindings/cs/rl.net/LiveModel.cs

```diff
@@ -49,10 +53,24 @@ public bool TryChooseRank(string actionId, string contextJson, RankingResponse r
             return result == NativeMethods.SuccessStatus;
         }
 
-        public bool TryReportOutcome(string actionId, float outcome, ApiStatus apiStatus = null)
+        public bool TryReportOutcome(string actionId, object outcome, ApiStatus apiStatus = null)
         {
-            int result = LiveModelReportOutcome(this.NativeHandle, actionId, outcome, apiStatus.ToNativeHandleOrNullptr());
+            string objStr = JsonConvert.SerializeObject(outcome);
+            int result;
```

Comment:
Why would we serialize the outcome just to reparse it? Wouldn't it make more sense to simply check the object for type float?

Reply:
float casting throws exception if int value is passed. Maybe I can do Convert.ToSingle which should handle both and then if it throws then do Convert.ToString